syntax = "proto3";
package fixbug;
option cc_generic_services = true;


message RegisterRequest { 
    string name = 1;
    string password = 2;
}
message RegisterResponse {
    bool success = 1;
    string msg = 2;
}
//rpc服务类,一个服务多个方法
service RegisterService {
    rpc Register(RegisterRequest) returns(RegisterResponse);
}


message LoginRequest { //登录请求
    string name = 1;
    string password = 2;
}
message LoginResponse { 
    bool success = 1;
}

message ChatRequest {
    string question = 1;
}
message ChatResponse {
    string answer = 1;
}

service LoginChatService {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Chat(ChatRequest) returns(ChatResponse);
}